cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -650,11 +650,13 @@ impl ASTNode<Option<cst::Cond>> {
             }
         };
 
-        match (maybe_is_when, maybe_expr) {
-            (Some(true), Some(e)) => Some(e),
-            (Some(false), Some(e)) => Some(construct_expr_not(e, src.clone())),
-            _ => None,
-        }
+        maybe_expr.map(|e| {
+            if maybe_is_when {
+                e
+            } else {
+                construct_expr_not(e, src.clone())
+            }
+        })
```

Comment:
I think this change results in a different program logic -- what happens when `maybe_expr` is `None`? The quick fix would be to add a question mark on line 651.

OTOH, given that the CI passed, perhaps this logic wasn't important. Or we're just not doing a good job of testing parse errors :)

Reply:
On my reading, the behavior when `maybe_expr` is `None` is unchanged -- `Option::map` returns `None` if the receiver is `None`

Reply:
Haha good point. I missed the `map`.